src/sparse/KokkosSparse_gauss_seidel.hpp

```diff
@@ -131,7 +131,8 @@ void block_gauss_seidel_symbolic(
                         is_graph_symmetric);
 }
 
-template <typename KernelHandle, typename lno_row_view_t_,
+template <KokkosKernels::SparseMatrixFormat format = KokkosKernels::BlockCRS,
```

Comment:
If BlockCRS is the default format, what happens when you pass in a regular CrsMatrix like in existing code? Does it just deduce the block size to be 1 and then run in the BlockCRS path?

Reply:
Top level `block_gauss_seidel_*` methods take `block_size` explicit argument and set it on the GS handle (e.g. see [here](https://github.com/kokkos/kokkos-kernels/blob/bcc505d321da561889285184a085220da8920515/src/sparse/KokkosSparse_gauss_seidel.hpp#L298)) before calling `gauss_seidel_*` methods, so the later fetch `block_size` from the handle instead of taking it as an explicit argument. It's how it worked - I didn't need to change how block size is passed.

`SparseMatrixFormat` tag is independent of block size and only tells how values are ordered. For bs=1 the ordering is identical for all formats. Maybe we should remove the default and require explicit tag for clarity (and add `CRS = BlockCRS` for consistency) ?

> On a side note, block size can be actually deduced with `auto bs = int_sqrt( values.extent(0) / entries.extent(0)`, but that works only for square blocks, so I keep passing explicit `block_size` around.

Reply:
@MikolajZuzek OK, that's fine then. I do like the idea of having a Crs tag as the default though, since I think users might be confused now if they're looking for the right functions to call for a regular Crs matrix.

Reply:
Added `CRS` format tag (it aliases `BlockCRS` to avoid unnecessary specializations) and made it default for non-block GS methods on e6c547030f68b099d5999b34c5eb08e82a987a83